_source/_posts/2022-02-23-angular-async-config.md

```diff
@@ -0,0 +1,340 @@
+
+```ts
+const routes: Routes = [
+  { path: 'profile', component: ProfileComponent }
+];
+
+@NgModule({
+  declarations: [ AppComponent, ProfileComponent ],
+  imports: [
+    BrowserModule,
+    RouterModule.forRoot(routes)
+  ],
+  bootstrap: [AppComponent]
+})
+export class AppModule { }
+```
+
+You might be surprised to see routing as an example of defining configuration directly in code but passing in application-wide configuration into a module's [`forRoot()` static method](https://angular.io/guide/ngmodule-faq#what-is-the-forroot-method) is precisely that. 
+
+If you've followed many of our code examples and blog posts to integrate Okta into Angular apps, you've followed a similar pattern where configuration is defined directly in the application. 
+
+Your configuration code looks something like this:
+
+```ts
+import { OktaAuthModule, OKTA_CONFIG } from '@okta/okta-angular';
+import { OktaAuth } from '@okta/okta-auth-js';
+
+const oktaAuth = new OktaAuth({
+  issuer: 'https://{yourOktaDomain}/oauth2/default',
+  clientId: '{yourClientId', 
+  redirectUri: window.location.origin + '/login/callback'
+});
+
+@NgModule({
+  declarations: [ AppComponent, ProfileComponent ],
+  imports: [
+    BrowserModule,
+    AppRoutingModule,
+    OktaAuthModule
+  ],
+  providers: [
+    { provide: OKTA_CONFIG, useValue: { oktaAuth } }
+  ],
+  bootstrap: [AppComponent]
+})
+export class AppModule { }
+```
+
+**Summary**
+
+The most straightforward way to add configuration to your app is when the configuration does not change based on external factors.
+
+**When to use**
+
+* As often as you can since it's the easiest way to configure things
+
+**Best for**
+
+* Static app configurations
+* Configuring third-party libraries
+* Quick tests
+
+**Watch out for**
+
+* Configuration that involves private keys or tokens
+
+## Configuration that changes by environment
+
+Angular has a built-in way to support per environment differences using the `environments/environment.*.ts` files. When serving locally, Angular CLI uses the values in `environments/environment.ts`, and when you build for prod, Angular CLI substitutes `environment.prod.ts` instead. You can see this file substitution defined in the `angular.json` build configuration. And if you have more environments to support, you can customize the build configuration to suit your needs.
+
+The environment files are helpful when you have different configurations you want to support at build time. Some examples include enabling user analytics only on prod environments or defining the API endpoints your QA environment calls.
+
+An example of configuration that changes by environment already built into your app is in `src/main.ts`. Here you see the following:
```

Comment:
```suggestion
`src/main.ts` is an example of a configuration that changes based on the environment already built into your app. Here you see the following: 
```
clarity, but does this still make sense?

Reply:
How about 
> `src/main.ts` contains an example of a configuration that changes based on the environment already built into your app.

I'll go ahead with this for now and can edit per your review suggestions.

Reply:
that looks great Alisa, thanks for refining - I think it's easier to read in active voice.